Source/Adobe.Target.Client/OnDevice/DecisioningDetailsExecutor.cs

```diff
@@ -0,0 +1,323 @@
+            return details.Match(
+                _ =>
+                {
+                    notification.Mbox = new NotificationMbox(globalMbox);
+                    return notification;
+                }, mboxRequest =>
+                {
+                    notification.Mbox = new NotificationMbox(mboxRequest.Name);
+                    return notification;
+                }, viewRequest =>
+                {
+                    notification.View = new NotificationView(viewRequest.Name, viewRequest.Key);
+                    return notification;
+                });
+        }
+
+        private static List<Metric> GetMetrics(object metricsObject)
+        {
+            return metricsObject is not JObject metricsJObject ? null : metricsJObject.ToObject<List<Metric>>();
+        }
+
+        private static void UnhandledResponse(RequestDetailsUnion details, PrefetchResponse prefetchResponse, ExecuteResponse executeResponse)
+        {
+            _ = details.Match<object>(
+                _ =>
+                {
+                    var response = new PageLoadResponse();
+                    if (prefetchResponse != null)
+                    {
+                        prefetchResponse.PageLoad = response;
+                        return null;
+                    }
+
+                    executeResponse.PageLoad = response;
+                    return null;
+                }, mboxRequest =>
+                {
+                    if (prefetchResponse != null)
+                    {
+                        var prefetchMboxResponse = new PrefetchMboxResponse(mboxRequest.Index, mboxRequest.Name);
+                        prefetchResponse.Mboxes ??= new List<PrefetchMboxResponse>();
+                        prefetchResponse.Mboxes.Add(prefetchMboxResponse);
+                        return null;
+                    }
+
+                    var response = new MboxResponse(mboxRequest.Index, mboxRequest.Name);
+                    executeResponse.Mboxes ??= new List<MboxResponse>();
+                    executeResponse.Mboxes.Add(response);
+                    return null;
+                }, viewRequest =>
+                {
+                    var view = new View();
+                    prefetchResponse.Views ??= new List<View>();
+                    prefetchResponse.Views.Add(view);
+                    return null;
+                });
+        }
+
+        private static bool PropertyTokenMismatch(IReadOnlyList<string> rulePropertyTokens, string propertyToken)
+        {
+            if (string.IsNullOrEmpty(propertyToken) || rulePropertyTokens == null || rulePropertyTokens.Count == 0)
+            {
+                return false;
+            }
+
+            return !rulePropertyTokens.Contains(propertyToken);
+        }
+
+        private static IReadOnlyList<OnDeviceDecisioningRule> GetDetailsRules(RequestDetailsUnion details, OnDeviceDecisioningRuleSet ruleSet)
+        {
+            return details.Match(
```

Comment:
Could we have proper methods instead of ternary operators.

Reply:
Fixed